twake/frontend/src/app/scenes/Setup/Setup.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react';
+
+import Emojione from 'components/Emojione/Emojione';
+import RouterServices from 'services/RouterServices';
+
+import './Setup.scss';
+
+type ReadyState =
+  | {
+      elasticsearch_connection: boolean;
+      elasticsearch_mapping: boolean;
+      db_connection: boolean;
+      db_mapping: boolean;
+      init: boolean;
+    }
+  | true;
```

Comment:
Does it means that it is true when every state above is true? You should not define types like this which are changing based on condition. Instead you have to have a `isReady()` function which tells you is it is ready based on the ReadyState.